app/assets/javascripts/uploadcare/widget/tabs/camera-tab.coffee

```diff
@@ -198,3 +200,26 @@ uploadcare.namespace 'widget.tabs', (ns) ->
 
       @__recorder.stop()
       @__chunks = []
+
+    __guessExtensionByMime: (mime) ->
+      known_containers = {
+        'mp4': 'mp4',
+        'ogg': 'ogg',
+        'webm': 'webm',
+        'quicktime': 'mov',
+        'x-matroska': 'mkv',
+      }
+      # e.g. "video/x-matroska;codecs=avc1,opus"
+      if mime
+        # e.g. ["video", "x-matroska;codecs=avc1,opus"]
+        mime = mime.split('/')
+        if mime[0] == 'video'
+          # e.g. "x-matroska;codecs=avc1,opus"
+          mime = mime.slice(1).join('/')
+          # e.g. "x-matroska"
+          container = mime.split(';')[0]
+          # e.g. "mkv"
+          if known_containers[container]
+            return known_containers[container]
+      # In all other cases just return the base extension for all times
+      return 'avi'
```

Comment:
Fallback to well known `avi` to prevent awful extension like `x-matroska;codecs=avc1,opus` in the future.